modules/codegen/src/main/scala/com/twilio/guardrail/Context.scala

```diff
@@ -1,7 +1,7 @@
 package com.twilio.guardrail
 
-case class Context(framework: Option[String], tracing: Boolean)
+case class Context(framework: Option[String], tracing: Boolean, http4sAuthedRoutes: Boolean)
```

Comment:
This PR represents a significant time investment, I appreciate what you've done so far. One of the design principles that may not have been clear is that the core of guardrail does not actually know anything about implementation details of any libraries; as a result, having a top-level context boolean that only applies to a single framework is undesirable.

An alternate strategy could be what we do in the [dropwizard generator](https://github.com/twilio/guardrail/blob/master/modules/codegen/src/main/scala/com/twilio/guardrail/generators/SwaggerGenerator.scala#L42-L44), where we build [`SecurityRequirements`](https://github.com/twilio/guardrail/blob/master/modules/codegen/src/main/scala/com/twilio/guardrail/terms/SwaggerTerm.scala#L47) both globally (set top-level in the OpenAPI specification) as well as locally (configured for an individual operation).

You could then use the presence or absence of `SecurityRequirements` to enable or disable this functionality.

Reply:
One other consideration here is that as this is a global boolean, there's no convenient way to combine authed and unauthed paths in the same configuration. By using this `security` context configuration, it could be turned on and off by route.

Reply:
I expected some feedback here. As you mention, it didn't fit so well with the rest of the design. The intention here was to not change the behavior of the existing solution if you didn't want to opt-in to it. I'll do another pass on it and see what I can find a better way to solve it.